cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -536,6 +554,17 @@ func main() {
 		RefreshCfgThrottle: refreshCfgThrottle,
 	}
 
+	if *performConnectivityTestsOnStartup == true {
+		for _, instanceConfig := range cfgs {
+			err := checkInstanceConnectivity(instanceConfig.Instance, instanceConfig.IsPublic, proxyClient);
+			if err != nil {
+				log.Fatal(err)
+			}
```

Comment:
Thoughts:

- what about timeouts?
- perform the tests concurrently? (fail on first failure)

Reply:
1. If it times out, it should fail I would imagine. That's what we want, right? 
2. You could; there's no reason you could not. I'm a bit new to Go; what's the best way to do this? Would I be doing something like this or is there something simpler? https://stackoverflow.com/a/24239322/1554844

It feels like something that is built into most other languages and is made very easy.

Reply:
I implemented 2. Let me know if there is a better way, it wasn't all that bad in Go actually.